plugins/imjournal/imjournal.c

```diff
@@ -520,24 +520,23 @@ persistJournalState(void)
 	DEFiRet;
 	FILE *sf = NULL; /* state file */
 	char tmp_sf[MAXFNAME];
-	size_t n;
 
 	DBGPRINTF("Persisting journal position, cursor: %s, at head? %d\n",
 			  journalContext.cursor, journalContext.atHead);
 
 	/* first check that we have valid cursor */
 	if (!journalContext.cursor) {
+		DBGPRINTF("Journal cursor is not valid, ok...\n");
 		ABORT_FINALIZE(RS_RET_OK);
 	}
 
 	/* we create a temporary name by adding a ".tmp"
 	 * suffix to the end of our state file's name
 	 */
-	n = snprintf(tmp_sf, sizeof(tmp_sf), "%s.tmp", cs.stateFile);
-	if (n >= sizeof(tmp_sf)) { /* backup in case of too long path name */
-		strncpy(tmp_sf, cs.stateFile, sizeof(tmp_sf) - 5);
-		strcat(tmp_sf, ".tmp");
-	}
+#define SUFFIX ".tmp"
+	snprintf(tmp_sf, sizeof(tmp_sf), "%.*s%*.s",
```

Comment:
While this, after lot of digging trough docs seems to do the trick, IMHO it warrants some explaining comment, preferably longer one ;-) 
Also, what is the advantage of doing the argument-defined minimum width with statically defined constant?

Reply:
I really smiled when it occurred to me that this would work well, so yes, I suppose a comment is in order. I will add that. As for the advantage of the minimum, I am not sure exactly why I left it there, it isn't necessary and I really should have minimized it in advance. I will remove it to save some typing in the comment. :)